crates/rpc/rpc-builder/src/auth.rs

```diff
@@ -68,6 +73,9 @@ where
 
     let fee_history_cache =
         FeeHistoryCache::new(eth_cache.clone(), FeeHistoryCacheConfig::default());
+
+    let initial_value: Option<(SealedBlock, Vec<Receipt>)> = None;
+    let (local_pending_block_watcher_tx, _) = watch::channel(initial_value);
```

Comment:
this can happen entirely inside the EthApi

a new receiver can be created on demand `Sender::subscribe`